client/engine/chainservice/adjudicator/challenge_test.go

```diff
@@ -104,7 +104,7 @@ func runChallengeWithTurnNum(t *testing.T, turnNum uint64, pc preparedChain) {
 	}
 
 	// Use random nonces so we can run this test multiple times against the same chain
-	nonce := uint64(rand.NewSource(time.Now().Unix()).Int63())
+	nonce := uint64(rand.GetRandGenerator().Int63())
```

Comment:
This is outside the scope of your changes but instead of using `Int63()` and casting to a `uint64` we can just call `Uint64()`

Reply:
I was wondering if the reason we are using the current approach is to fit into some evm constraint. Sounds like maybe that's not the case...

Reply:
264d6f7c